kernel/src/debug.rs

```diff
@@ -555,6 +557,16 @@ pub fn debug_println(args: Arguments) {
     writer.publish_bytes();
 }
 
+pub fn debug_slice(slice: &ReadableProcessSlice) {
+    let writer = unsafe { get_debug_writer() };
+
+    for b in slice.iter() {
+        let buf: [u8; 1] = [b.get(); 1];
+        let _ = writer.write(&buf);
+    }
```

Comment:
We could make this a bit more idiomatic by changing the definition of write() to take a `ReadableProcessSlice` instead of `&[u8]`, since any `&[u8]` can be cast for free to a `ReadableProcessSlice`

Reply:
Maybe? This seems a bit backwards. It would mean that regular `debug!` calls of kernel information are cast into a `ReadableProcessSlice`? I agree that it makes this particular bit of code more idiomatic, but it would be a strange hoop to jump through and be confusing to most readers of `Debug`.

Reply:
Yeah, I don't feel too strongly about it. Worth nothing that even in the final binary, this does not get optimized to something more efficient -- there is actually a unique call to `write()` for every byte in the slice:

```
   1b336:       f011 fdcb       bl      2ced0 <_ZN6kernel5debug16get_debug_writer17h9eb5ff7542767abcE>
   1b33a:       f1a7 0919       sub.w   r9, r7, #25
   1b33e:       4605            mov     r5, r0
   1b340:       b156            cbz     r6, 1b358 <_ZN8capsules9print_log17PrintLog$LT$A$GT$4send17h5a74adc2852bad0fE+0x92>
   1b342:       f814 0b01       ldrb.w  r0, [r4], #1
   1b346:       4649            mov     r1, r9
   1b348:       f807 0c19       strb.w  r0, [r7, #-25]
   1b34c:       4628            mov     r0, r5
   1b34e:       2201            movs    r2, #1
   1b350:       f011 fb3e       bl      2c9d0 <_ZN76_$LT$kernel..debug..DebugWriterWrapper$u20$as$u20$kernel..debug..IoWrite$GT$5write17hbb6dbc4f34f5e269E>
   1b354:       3e01            subs    r6, #1
   1b356:       e7f3            b.n     1b340 <_ZN8capsules9print_log17PrintLog$LT$A$GT$4send17h5a74adc2852bad0fE+0x7a>
   1b358:       4628            mov     r0, r5
   1b35a:       f011 fddf       bl      2cf1c <_ZN6kernel5debug18DebugWriterWrapper13publish_bytes17hddce97158654c963E>
```